src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -130,4 +133,48 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
       case None => List.empty
     }
   }
+
+  def getEntitySubtrees(workspaceNamespace: String, workspaceName: String, entityType: String, entityNames: Seq[String], txn: RawlsTransaction): Seq[Entity] = txn withGraph { db =>
+    def nameFilter = new PipeFunction[Vertex, java.lang.Boolean] {
+      override def compute(v: Vertex) = entityNames.contains(v.getProperty("_name"))
+    }
+
+    def isLeaf = new PipeFunction[LoopPipe.LoopBundle[Vertex], java.lang.Boolean] {
+      override def compute(bundle: LoopPipe.LoopBundle[Vertex]): java.lang.Boolean = { true }
+    }
+
+    def emitAll = new PipeFunction[LoopPipe.LoopBundle[Vertex], java.lang.Boolean] {
+      override def compute(bundle: LoopPipe.LoopBundle[Vertex]): java.lang.Boolean = { true }
+    }
+
+    val subtreeEntities = getWorkspaceVertex(db, workspaceNamespace, workspaceName) match {
+      case Some(w) => {
+        val topLevelEntities = new GremlinPipeline(w).out(entityType).filter(nameFilter).iterator().map(loadEntity(_, workspaceNamespace, workspaceName)).toList
+        val nextLevelEntities = new GremlinPipeline(w).out(entityType).filter(nameFilter).as("outLoop").out().dedup().loop("outLoop", isLeaf, emitAll).iterator().map(loadEntity(_, workspaceNamespace, workspaceName)).toList
```

Comment:
Isn't this getting all sub-entities, not just the next level? If so the name is a bit misleading. Ditto for ```isLeaf```.

Also, is there some way to emit the ```topLevelEntities``` as an intermediate result without needing a separate pipe?